apps/artemis-web/utils/token.tsx

```diff
@@ -112,6 +112,46 @@ export const statuses = {
   Outdated: 'dark',
 };
 
+export const compareObjects = (obj1, obj2) => {
+  if (obj1 === obj2) return true;
+
+  if (
+    typeof obj1 !== 'object' ||
+    typeof obj2 !== 'object' ||
+    obj1 == null ||
+    obj2 == null
+  ) {
+    return false;
+  }
+
+  const keysA = Object.keys(obj1);
+  const keysB = Object.keys(obj2);
+
+  if (keysA.length !== keysB.length) {
+    return false;
+  }
+
+  let result = true;
+
+  keysA.forEach((key) => {
+    if (!keysB.includes(key)) {
+      result = false;
+    }
+
+    if (typeof obj1[key] === 'function' || typeof obj2[key] === 'function') {
+      if (obj1[key].toString() !== obj2[key].toString()) {
+        result = false;
+      }
+    }
+
+    if (!compareObjects(obj1[key], obj2[key])) {
```

Comment:
I do not get this, is it recursive? (it seems to be contained in the body of the `compareObjects` definition)

Reply:
yes it is. it performs a deep equality check of JS objects

Reply:
Thank you for the explanation Alex. Resolving.